### PR TITLE
Bound ISourceControlProvider to GitProvider

### PR DESCRIPTION
**Pull request by alanturing88**
Hi,

I'm working on issue #2353 

With this change it is possible to create a repo, but I am still not able to commit...

I will continue later on this.

Regards.

### CODE REVIEW

**Comment by Hosch250** [comment]
So, you hard-coded the type to be injected? Now this is less ready to support other SC systems, like SVN. Whatever is going on, this isn't the solution.

**Comment by Hosch250** [comment]
I'm not sure what's causing this problem, but I'll pull the latest version and see if I can reproduce it soon.

**Comment by retailcoder** [comment]
@Hosch250 no, the line `Bind<GitProvider>().ToSelf().InSingletonScope();` was removed, and the fix is consistent with the `ActivationException` documented in the linked issue.

**Comment by Hosch250** [comment]
Yes, but the fix should be to bind the GitProvider to the ISourceControlProvider, I *think*.

**Comment by Hosch250** [comment]
Ah, I'm blind.

File: RetailCoder.VBE/Root/RubberduckModule.cs [hunk 1]
```diff
@@ -64,7 +64,8 @@ public override void Load()
             Bind<Sinks>().ToSelf().InSingletonScope();
             Bind<App>().ToSelf().InSingletonScope();
             Bind<RubberduckParserState>().ToSelf().InSingletonScope();
-            Bind<GitProvider>().ToSelf().InSingletonScope();
+            Bind<ISourceControlProvider>().To<GitProvider>();
```
Comment (retailcoder) [inline RetailCoder.VBE/Root/RubberduckModule.cs]: There's a problem with that though: when a scope isn't specified, Ninject uses `TransientScope`, which doesn't dispose of `IDisposable` dependencies.

    public class GitProvider : SourceControlProviderBase, IDisposable

And `GitProvider` implements `IDisposable`, so `TransientScope` is going to cause issues one way or another.

Comment (alanturing88) [inline RetailCoder.VBE/Root/RubberduckModule.cs]: Ok,

I will add `.InCallScope()`. think that's the best way.

Comment (alanturing88) [inline RetailCoder.VBE/Root/RubberduckModule.cs]: Ok,

I will add `.InCallScope()`. think that's the best way.

**Comment by alanturing88** [comment]
Ok,

I will add `.InCallScope()`. think that's the best way.

**Pull request merged**